### ISSUE TITLE
[Suggestion] Potential Method for non-append system?

### ISSUE THREAD

**Issue by Merith-TK** [darenliang/dsfs#8]
Why not go back to something like DSCLI a little?

Why not make the bot use an "forum channel" like this, 
![image](https://user-images.githubusercontent.com/10422110/231706240-241e0b56-d444-4104-9eb8-486f7d463bf9.png)
Where the "initial message" contains file information, and then is uploaded in chunks?

or is there not yet discord bot mappings for that?

**Comment by Merith-TK** [comment]
Have the "forum title" be the file name, its max character limit is 100, so maybe 
```
title: filename.dat
postMSG: <contain file information data>
replies: <the file data>
```

**Comment by darenliang** [comment]
One of the main motivators for dsfs is that it addresses the shortcomings of dscli.

Filenames no longer have a limit and there is support for an arbitrary number of files and folders, along with fuse support.

I'm afraid it isn't possible to use the same system dscli uses anymore. If you had files stored via dscli you can move it over to dsfs or continue to access the files via dscli.

The transaction formats for dsfs aren't stable at the moment so there could be backwards incompatible changes but I will find a way to address that. Discord recently increased the file limit from 8MB to 25MB but dsfs can't safely use the new file limit size without breaking existing dsfs servers.

**Comment by Merith-TK** [comment]
Add a new tx record tag for items that's sole purpose is to say this file
uses the new file size limit, that should do it?

**Comment by darenliang** [comment]
Yes that's a good approach. To make it more future proof I think it would be a good idea to add a tx version number and write migration code to automatically convert old versions to newer versions.
